packages/react-cookie/src/manager.ts

```diff
@@ -0,0 +1,65 @@
+import cookie from 'cookie';
+import {hasDocumentCookie} from './utilities';
+import {Cookies, CookieValue} from './types';
+
+export {CookieContext} from './context';
+
+export class CookieManager {
+  private readonly cookies = new Map<string, CookieValue>();
+
+  constructor(cookies: Cookies | string = {}) {
+    const parsedCookies =
+      typeof cookies === 'string' ? cookie.parse(cookies) : cookies;
+
+    Object.entries(parsedCookies).forEach(([key, value]) => {
+      this.setCookie(key, value);
+    });
+  }
+
+  getCookie(name: string) {
+    const cookie = this.cookies.get(name);
+
+    return cookie && cookie.value;
+  }
+
+  getCookies() {
+    const cookies: Cookies = {};
+
+    for (const [key, value] of this.cookies) {
+      cookies[key] = value;
+    }
+
+    return {
+      ...cookies,
+    };
+  }
+
+  setCookie(name: string, value: string | CookieValue) {
```

Comment:
I think realistically on the server we want to set the cookie using a `networkManager` form `@shopify/react-network`,  which would probable entail

- depending on it here and passing it in from the component, or rolling this back into that library
- adding a `setCookie` property on the network manager
- adding logic to `applyToContext` that calls `ctx.cookies.set` (which will handle setting `set-cookie` as a user would expect)